l1-contracts/test/merkle/Frontier.t.sol

```diff
@@ -8,7 +8,10 @@ import {NaiveMerkle} from "./Naive.sol";
 import {FrontierMerkle} from "../harnesses/Frontier.sol";
 
 contract FrontierTest is Test {
-  function setUp() public {}
+  function setUp() public {
+    // Pause gas metering as calculating the root on each insert is expensive
+    vm.pauseGasMetering();
```

Comment:
Interesting, have not seen this before

Reply:
doing it 1000 times in a loop in this test, for each tree. Was making it exceed limits. I assumed its fine to just not meter as we dont use it like this in prod.